### PR TITLE
Fixed sorting collection when watchable property changes

### PR DESCRIPTION
**Pull request by herkyl**
Collection was not getting sorted when a watchable property changed while comparator was a function

### CODE REVIEW

**Comment by herkyl** [comment]
Tests work locally.

File: ampersand-filtered-subcollection.js [hunk 1]
```diff
@@ -356,9 +356,8 @@ assign(FilteredCollection.prototype, Events, {
         }
 
         //If we were asked to sort, or we aren't gonna get a sort later and had a sortable property change
-        if (
-            action === 'sort' ||
-            (propName && !sortable && includes([this.comparator, this.collection.comparator], propName))
+        if (action === 'sort' ||
+            (propName && !sortable && includes([this.comparator, this.collection.comparator].concat(this._watched), propName))
```
Comment (wraithgar) [inline ampersand-filtered-subcollection.js]: I don't think you need this concat here, I think you can just add `this._watched` to the array that is being checked against propName

Comment (herkyl) [inline ampersand-filtered-subcollection.js]: @wraithgar `this._watched` is an array. How would you go about adding it?

Comment (wraithgar) [inline ampersand-filtered-subcollection.js]: Oh right, I missed that somehow.  Carry on.

**Comment by wraithgar** [comment]
Was there a reason you refactored two tests instead of adding new ones for this new use case?  Sort of feels like we're losing coverage for cases where watched isn't set now.

**Comment by herkyl** [comment]
@wraithgar I think you are right, I will separate those tests

**Comment by wraithgar** [comment]
Ok great thank you for making this PR. Seems fine to me. +1

**Comment by herkyl** [comment]
Whats the status on this?

**Comment by wraithgar** [comment]
Was waiting on another +1 but I think a week's enough time to wait.

Can you revert the version in package.json and then I'll merge this.  (The version bump is part of a separate publish process already)

**Comment by herkyl** [comment]
Done

**Comment by wraithgar** [comment]
Published as `v2.0.5`

**Comment by wraithgar** [comment]
Thank you!

**Pull request merged**